tests/input_flows.py

```diff
@@ -0,0 +1,1283 @@
+"""
+Central place for defining all input flows for the device tests.
+
+Each model has potentially its own input flow, and in most cases
+we need to distinguish between them. Doing it at one place
+offers a better overview of the differences and makes it easier
+to maintain. The whole `device_tests` folder can then focus
+only on the actual tests and data-assertions, not on the lower-level
+input flow details.
+"""
+
+import time
+from typing import Callable, Generator, Optional
+
+from trezorlib import messages
+from trezorlib.debuglink import (
+    DebugLink,
+    LayoutContent,
+    TrezorClientDebugLink as Client,
+    multipage_content,
+)
+
+from . import buttons
+from .common import (
+    check_PIN_backoff_time,
+    click_info_button,
+    click_through,
+    read_and_confirm_mnemonic,
+    recovery_enter_shares,
+)
+
+GeneratorType = Generator[None, messages.ButtonRequest, None]
+
+B = messages.ButtonRequestType
+
+
+def swipe_if_necessary(
+    debug: DebugLink, br_code: Optional[messages.ButtonRequestType] = None
+) -> GeneratorType:
+    br = yield
+    if br_code is not None:
+        assert br.code == br_code
+    if br.pages is not None:
+        for _ in range(br.pages - 1):
+            debug.swipe_up()
+
+
+class InputFlowBase:
+    def __init__(self, client: Client):
+        self.client = client
+        self.debug: DebugLink = client.debug
+        self.layout = client.debug.wait_layout
+
+    def model(self) -> Optional[str]:
+        return self.client.features.model
+
+    def get(self) -> Callable[[], GeneratorType]:
+        self.client.watch_layout(True)
+
+        # There could be one common input flow for all models
+        if hasattr(self, "input_flow_common"):
+            return getattr(self, "input_flow_common")
+        elif self.model() == "T":
+            return self.input_flow_tt
+        elif self.model() == "R":
+            return self.input_flow_tr
+        else:
+            raise ValueError("Unknown model")
+
+    def input_flow_tt(self) -> GeneratorType:
+        """Special for TT"""
+        raise NotImplementedError
+
+    def input_flow_tr(self) -> GeneratorType:
+        """Special for TR"""
+        raise NotImplementedError
+
+
+class InputFlowSetupDevicePINWIpeCode(InputFlowBase):
```

Comment:
Allow me one last pushback before "decentralizing it" completely - as I started doing it, things started appearing that I really do not like. Mostly for slip39/bip39 input flows, the results just do not look good IMO.

I have done some of the "decentralization" in [c62d408](https://github.com/trezor/trezor-firmware/pull/2803/commits/c62d40849fc2a265d3a29d8c2ec562277819bcb4) and it uncovered for example these issues/unneeded complications:
- some input flows are used by more than one test-case, so they would need to be "centralized" anyway so we can reuse them
- the amount of imports that each device test needs to do for defining input flow is quite big - it needs to know a lot of details of how to click the button, how to swipe, how to assert ButtonRequestType etc.
- the result is then mixed of importing some InputFlows from common, and defining some locally

As one who spends a lot of time with device tests, having everything in `input_flows.py` is my preferred way, it can reuse all things/imports in this one file, and device tests do not have to care about any of this.

![image](https://user-images.githubusercontent.com/42543243/233948142-b3a9d180-d513-4f92-a22d-d9b704a70d5c.png)

The two-window view above (which everybody can do in VSCode by just clicking the InputFlowName) seems to me like an ideal situation - you see the high-level test-details on the left in a test file, while you can see/update the low-level input-flow-details on the right